contributors/design-proposals/core-metrics-pipeline.md

```diff
@@ -0,0 +1,315 @@
+# Core Metrics Pipeline in kubelet
+
+**Author**: David Ashpole (@dashpole)
+
+**Last Updated**: 12/21/2016
+
+**Status**: Draft Proposal (WIP)
+
+This document proposes a design for an internal Core Metrics Pipeline.
+
+<!-- BEGIN MUNGE: GENERATED_TOC -->
+
+- [Core Metrics Pipeline in kubelet](#core-metrics-pipeline-in-kubelet)
+  - [Introduction](#introduction)
+    - [Background](#background)
+    - [Motivations](#motivations)
+    - [Proposal](#proposal)
+    - [Non Goals](#non-goals)
+  - [Design](#design)
+    - [Proposed Core Metrics API:](#proposed-core-metrics-api)
+    - [Core Machine Info:](#core-machine-info)
+  - [Implementation Plan](#implementation-plan)
+  - [Rollout Plan](#rollout-plan)
+  - [Implementation Status](#implementation-status)
+
+<!-- END MUNGE: GENERATED_TOC -->
+
+## Introduction
+
+### Background
+The [Monitoring Architecture](https://github.com/kubernetes/kubernetes/blob/master/docs/design/monitoring_architecture.md) proposal contains a blueprint for a set of metrics referred to as "Core Metrics".  The purpose of this proposal is to specify what those metrics are, and how they will be collected on the node.
```

Comment:
> "resource-metrics", "system-resource-metrics", "utilization-metrics"

So, the name of the API served to expose these metrics to Kubernetes components (the HPA controller, the scheduler in the future) is the "resource metrics API" (also known as the "master metrics API", but that's a bit of a misnomer).  I, personally, would lean towards the "system resource metrics API", "kubelet metrics API", or something along those lines, to distinguish it from the "resource metrics API".